Mathlib/CategoryTheory/Presentable/CardinalFilteredPresentation.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Presentable.Limits
+
+/-!
+# Presentable generators
+
+Let `C` be a category and `κ` a regular cardinal in a universe `w`.
+In this file, we introduce a structure `CardinalFilteredPresentation X κ`
+which consists of a presentation of an object `X : C` as a colimit of a functor
+from a `κ`-filtered category. This is used in order to introduce a property
+`h : AreCardinalFilteredGenerators G κ` of a family `G : ι → C` on objects:
+this property says that the objects `G i` are all `κ`-presentable and that
+any object in `C` identifies as a `κ`-filtered colimit of these objects.
+We show in the lemma `AreCardinalFilteredGenerators.presentable` that it
+follows that any object `X` is presentable (relatively to a possibly
+larger regular cardinal `κ'`).
+
+Finally, we define a typeclass `HasCardinalFilteredGenerators C κ` saying
+that `C` is locally `w`-small and that there exists a family `G : ι → C`
+indexed `ι : Type w` such that `AreCardinalFilteredGenerators G κ` holds.
+This is used in the definition of locally presentable and locally accessible
+categories in the file `CategoryTheory.Presentable.LocallyPresentable`.
+
+## References
+* [Adámek, J. and Rosický, J., *Locally presentable and accessible categories*][Adamek_Rosicky_1994]
+
+-/
+
+universe w v u
+
+namespace CategoryTheory
+
+open Limits
+
+section
+
+variable {C : Type u} [Category.{v} C]
+
+/-- Given an object `X` in a category `C` and a regular
+cardinal `κ` (in the universe `w`), this is a presentation of `X` as the colimit
+of a functor `J ⥤ C` where `J` is a `κ`-filtered category. -/
+structure CardinalFilteredPresentation (X : C) (κ : Cardinal.{w}) [Fact κ.IsRegular] where
+  /-- the index category of the presentation -/
+  J : Type w
+  /-- the category structure on the index type -/
+  category : Category.{w} J := by infer_instance
+  isCardinalFiltered : IsCardinalFiltered J κ := by infer_instance
+  /-- the functor that is part of the presentation -/
+  F : J ⥤ C
+  /-- the `ι` field of the cocone -/
+  ι : F ⟶ (Functor.const _).obj X
+  /-- `X` is the colimit of the functor `F` -/
+  isColimit : IsColimit (Cocone.mk _ ι)
+
```

Comment:
```suggestion

initialize_simps_projections CardinalFilteredPresentation (-category, -isCardinalFiltered)

```

This way you don’t need to manually specify projections like you do in `CardinalFilteredPresentation.ofIsColimit`.